config/src/main/java/com/typesafe/config/impl/ConfigDocumentParser.java

```diff
@@ -106,9 +116,13 @@ private boolean checkElementSeparator(Collection<AbstractConfigNode> nodes) {
                 while (true) {
                     if (Tokens.isIgnoredWhitespace(t) || isUnquotedWhitespace(t)) {
                         //do nothing
+                    } else if (Tokens.isComment(t)) {
+                        nodes.add(new ConfigNodeComment(t));
+                        t = nextToken();
+                        continue;
```

Comment:
this is a little clunky to sometimes return, sometimes continue, and sometimes fall down to add ConfigNodeSingleToken.
I'd suggest putting a nodes.add in every branch, and then for those that don't return always fall down to `t = nextToken`
(i.e. remove `t = nextToken` and `continue` from this branch, and add `nodes.add(new ConfigNodeSingleToken)` to the two branches that don't continue or return).